File: portal/src/main/resources/static/service/appointment/appointmentService.js
function appointmentService(MODULE_CONFIG, $http) {
	
	this.addAppointment = function(appointment) {
	return 	$http.post(MODULE_CONFIG.ADDAPPOINTMENT(),appointment);
	};
	this.getAllTherpistByDepartment=function(id){
		return $http.get(MODULE_CONFIG.GETALLTHERAPISTSBYDEPARTMENT(id));
	};
	this.getAppointmentssByRole=function(adminUserName){
		return $http.get(MODULE_CONFIG.GET_APPOINTS_BY_ROLE(adminUserName));
	};
	this.getAppointmentssByRoleByPagination=function(adminUserName, page, size){
		return $http.get(MODULE_CONFIG.GET_APPOINTS_BY_ROLE_PAGE(adminUserName, page, size));
	};
	this.updateAppointment = function(appointment) {
		return 	$http.put(MODULE_CONFIG.UPDATEAPPOINTMENT(),appointment);
    };
    this.dropAppointment=function(id){
		return $http.delete(MODULE_CONFIG.DELETEAPPOINTMENT(id));
	};
	this.getListBetweenDates = function(appointments,page,size) {
		return 	$http.post(MODULE_CONFIG.GETAPPOINTMENTLISTBYBETWEENDATES( page, size),appointments);
    };
    
    this.getSubAppointmentssByRole=function(adminUserName){
		return $http.get(MODULE_CONFIG.GET_ALL_SUB_APPOINTS_BY_ROLE(adminUserName));
	};
	this.getSubAppointmentssByRoleforCalendarView=function(adminUserName){
		return $http.get(MODULE_CONFIG.GET_ALL_SUB_APPOINTS_BY_ROLE_FOR_CALENDARVIEW(adminUserName));
	};
	this.getTodaySubAppointments=function(name){
		return $http.get(MODULE_CONFIG.GET_TODAY_SUBAPPOINTMENTS(name));
	};
	this.updateSubappointmentStatus=function(id,status){
		return $http.post(MODULE_CONFIG.UPDATESUBAPPOINTMENTSTATUS(id,status));
	};
	this.getSubAppointmentsByInvoice = function(appointment) {
		return 	$http.post(MODULE_CONFIG.GETSUBAPPOINTMENTS_BYINVOICE(),appointment);
	};
    
	this.getAppointmentssByAdminByPagination=function(adminUserName, page, size){
		return $http.get(MODULE_CONFIG.GET_APPOINTS_BY_ADMIN_PAGE(adminUserName, page, size));
	};
	
	this.getAllSubAppointmentsByAppointment=function(id){
		return $http.get(MODULE_CONFIG.GET_ALL_SUBAPPOINTMENTS_BY_APPOINTMENT(id));
	};
	
	this.getSubAppointmentsByInvoiceNo=function(itemDto){
		return $http.post(MODULE_CONFIG.GET_SUBAPPOINTMENTS_BY_INVOICENO(),itemDto);
	};
	this.updateAppointmentWithAssignedTherapist=function(scheduleDtoForCalendar){
		return $http.post(MODULE_CONFIG.UPDATE_APPOINTMENT_WITH_ASSIGNEDTHERAPIST(),scheduleDtoForCalendar);
	};
	this.getSubAppointmentsByInvoicePage=function(appointment,page, size) {
		return 	$http.post(MODULE_CONFIG.GETSUBAPPOINTMENTS_BYINVOICE_PAGE(page, size),appointment);
	};
}

angular.module('HealthApplication').service('appointmentService', appointmentService);

File: hospital/src/main/java/com/openspace/HospitalMgnt/Therapist/Appointment/AppointmentService.java
package com.openspace.HospitalMgnt.Therapist.Appointment;

import java.util.List;

import org.springframework.data.domain.Page;

import com.openspace.Model.DoctorManagement.Appointment;
import com.openspace.Model.DoctorManagement.AppointmentInvoice;
import com.openspace.Model.DoctorManagement.SubAppointment;
import com.openspace.Model.DoctorManagement.SubAppointmentStatus;
import com.openspace.Model.Dto.AppointmentDto;
import com.openspace.Model.Dto.AppointmentInvoiceDto;
import com.openspace.Model.Dto.AppointmentsDateDto;
import com.openspace.Model.Dto.InvoiceSubAppointmentDto;
import com.openspace.Model.Dto.ItemDto;
import com.openspace.Model.Dto.PatientInvoiceDto;
import com.openspace.Model.Dto.ScheduleDtoForCalendar;
import com.openspace.Model.Dto.SubAppointmentDto;

public interface AppointmentService {
	void saveAppointment(AppointmentDto appoointmentDto);

	Page<SubAppointment> getAppointmentssByRole(String adminUserName, int page, int size);

	List<Appointment> getAppointmentssByRole(String adminUserName);

	int getAllAppointmentsCountByRole(String adminUserName);

	// void updateAppointment(AppointmentDto appoointmentDto);

	void updateAppointment(Appointment appoointmentDto);

	void deleteAppointment(Long id);

	Page<SubAppointment> getAppointlistByDates(AppointmentsDateDto appointmentsDateDto, int page, int size);

	List<SubAppointmentDto> getSubAppointmentssByRole(String adminUsername);

	List<SubAppointmentDto> getSubAppointmentssByRoleForCalendarView(String adminUserName);

	void updateSubappointmentStatus(Long id, SubAppointmentStatus status);

	List<SubAppointmentDto> getTodayAppointments(String adminUserName);
	
	List<PatientInvoiceDto> getSubAppointmentsByInvoice(InvoiceSubAppointmentDto appointmentDto );

	List<PatientInvoiceDto> getAllSubAppointmentsByInvoice(String username);

	Page<PatientInvoiceDto> getAllSubAppointmentsByInvoicePage(String username, int page, int size);

	Page<Appointment> getAppointmentsByRole(String adminUserName, int page, int size);

	List<SubAppointment> getAllSubAppointmentsByAppointment(Long id);

	List<ItemDto> getSubAppointmentsByInvoiceNo(AppointmentInvoiceDto invoiceNo);

	Page<Appointment> findAllTherapistAppointments(Long id, int page, int size);

	void updateAppointmentWithAssignedTherapist(ScheduleDtoForCalendar scheduleDtoForCalendar);

	Page<PatientInvoiceDto> getSubAppointmentsByInvoicePage(InvoiceSubAppointmentDto appointmentDto, int page,int size);
			
}
